src/Microsoft.AspNetCore.Hosting/Internal/HostingApplicationData.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    public class HostingApplicationData : IHostingApplicationData
```

Comment:
Why did you structure this type differently for AspNetCore vs Extensions? E.g. for AspNetCore the initialization code is in the type, but for Extensions it's in the calling type.

Reply:
I'm following the exact same pattern as IHostingEnvironment/HostingEnvironment. They are structured differently between AspNetCore.Hosting and Extensions.Hosting.